core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -412,58 +411,96 @@ public boolean entryExpiredInMemoryFromIteration(InternalCacheEntry<K, V> entry,
+      protected Boolean addTargetNotFound(Address sender) {
+         throw OutdatedTopologyException.RETRY_NEXT_TOPOLOGY;
       }
 
-      // Need to gather last access times
-      RetrieveLastAccessCommand rlac = cf.running().buildRetrieveLastAccessCommand(key, value, segment);
-      rlac.setTopologyId(topology.getTopologyId());
-
-      // In scattered cache read owners will only contain primary
-      return rpcManager.invokeCommand(info.readOwners(), rlac, new MaxResponseCollector<>(access),
-            rpcManager.getSyncRpcOptions()).toCompletableFuture();
+      @Override
+      protected Boolean addException(Address sender, Exception exception) {
+         if (exception instanceof CacheException) {
+            throw (CacheException) exception;
+         }
+         throw new CacheException(exception);
+      }
    }
 
-   static class MaxResponseCollector<T extends Comparable<T>> extends ValidResponseCollector<T> {
-      T highest;
-
-      MaxResponseCollector(T highest) {
-         this.highest = highest;
-      }
+   private static class ScatteredTouchResponseCollector extends AbstractTouchResponseCollector {
 
       @Override
-      public T finish() {
-         return highest;
+      public Boolean finish() {
+         // No other node was touched
+         return Boolean.FALSE;
       }
 
       @Override
-      protected T addValidResponse(Address sender, ValidResponse response) {
-         T value = (T) response.getResponseValue();
-         if (trace) {
-            log.tracef("Received response %s from %s when requesting access time for max idle", value, sender);
-         }
-         if (value != null && (highest == null || highest.compareTo(value) < 0)) {
-            highest = value;
+      protected Boolean addValidResponse(Address sender, ValidResponse response) {
+         Boolean touched = (Boolean) response.getResponseValue();
+         if (touched == Boolean.TRUE) {
+            // Return early if any node touched the value - as SCATTERED only exists on a single backup!
+            // TODO: what if the read was when one of the backups or primary died?
+            return Boolean.TRUE;
          }
          return null;
       }
+   }
 
+   private static class TouchResponseCollector extends AbstractTouchResponseCollector {
       @Override
-      protected T addTargetNotFound(Address sender) {
-         // We don't care about a node leaving
-         return null;
+      public Boolean finish() {
+         // If all were touched, then the value isn't expired
+         return Boolean.TRUE;
       }
 
       @Override
-      protected T addException(Address sender, Exception exception) {
-         throw ResponseCollectors.wrapRemoteException(sender, exception);
+      protected Boolean addValidResponse(Address sender, ValidResponse response) {
+         Boolean touched = (Boolean) response.getResponseValue();
+         if (touched == Boolean.FALSE) {
+            // Return early if any value wasn't touched!
```

Comment:
which scenario? this would be quite common when using max idle with scattered cache.

Reply:
This is the disc response collector, not the scattered one.

I was thinking it would be uncommon to have an operation that tries to expire an entry and finds another operation already expired/updated/removed the entry, because expirations themselves shouldn't be very common.

Reply:
To clarify this is ran on access, not expiration - so it would increase the frequency a bit more.

The other benefit of returning early is technically the collector is stateless, which would allow for a singleton impl, I just forgot to do that.

Reply:
True, you don't need two concurrent expirations to get a `false`, just one expiration and a regular read, but still I'm not 100% sold on having those touch commands floating around.